lib/reline/line_editor.rb

```diff
@@ -509,6 +512,7 @@ def render
   # by calculating the difference from the previous render.
 
   private def render_differential(new_lines, new_cursor_x, new_cursor_y)
+    Reline::IOGate.disable_auto_linewrap(true) if Reline::IOGate.win?
```

Comment:
Why is this needed? I think line_editor does not print line longer than terminal width.
(Maybe the case line_width == terminal_width case?)

If this is needed, moving it to `Reline::Windows#prep` and `Reline::Windows#deprep` is better

Reply:
The restore in deprep does not work because render_finished outputs a single, long line.
Windows console has mode flags. (https://learn.microsoft.com/en-us/windows/console/high-level-console-modes)
If ENABLE_VIRTUAL_TERMINAL_PROCESSING is set, cursor moves as expected by reline.
In environments where the flag is off or not supported, ENABLE_WRAP_AT_EOL_OUTPUT is on normally, causing a forced line break at the end of a line of output.
For example, outputting ```123456``` on a 6-column wide terminal would result in the following.
```
123456
_ #=> cursor moves next line
```
If ENABLE_WRAP_AT_EOL_OUTPUT is turned off, no forced line breaks occur. Instead, the cursor stays at the end of the line and subsequent characters overwrite the end of the line.
For example, outputting ```12345678``` on a 6-column wide terminal would result in the following.
```
123458 #=> cursor stays last column of line
```
Therefore, it was necessary to insert control of ENABLE_WRAP_AT_EOL_OUTPUT in the appropriate place.
Another reason was that if there was console output in the background, long lines could be destroyed if the flag was not restored while waiting for keystrokes.

Reply:
It makes sense. Thank you for the description :+1:

Reply:
done in df7054193d53d0b55f7787d2420af59c284c64c8 .